src/profile-logic/marker-data.js

```diff
@@ -160,19 +160,23 @@ export function extractMarkerDataFromName(
 
   // Match: "Bailout_MonitorTypes after add on line 1013 of self-hosted:1008"
   // Match: "Bailout_TypeBarrierO at jumptarget on line 1490 of resource://devtools/shared/base-loader.js -> resource://devtools/client/shared/vendor/immutable.js:1484"
+  // Match: "Bailout_ShapeGuard at jumptarget on line 7 of moz-extension://<URL>"
   const bailoutRegex =
     // Capture groups:
-    //       type   afterAt    where        bailoutLine  script functionLine
-    //        ↓     ↓          ↓                  ↓        ↓    ↓
-    /^Bailout_(\w+) (after|at) ([\w _-]+) on line (\d+) of (.*):(\d+)$/;
+    //       type   afterAt    where        bailoutLine  script     functionLine
+    //        ↓     ↓          ↓                  ↓        ↓        ↓
+    /^Bailout_(\w+) (after|at) ([\w _-]+) on line (\d+) of (.*?)(?::(\d+))?$/;
+  //                                                               ↑
+  // Colon in non-capturing group which only matches when functionLine is present
```

Comment:
I know this regexp is starting to get confusing. Another approach is to put everything after "of" into its own capture group, and then parse that part later in a second step. Since I've already written this and it works, I kept it for review. Let me know what you think, I'm open for changing it.